src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloadsTestBase.cs

```diff
@@ -1,63 +1,286 @@
         {
             var test = new VerifyVB.Test
             {
-                TestCode = source,
+                TestCode = originalSource,
                 ReferenceAssemblies = version,
             };
 
+            if (!string.IsNullOrEmpty(fixedSource))
+            {
+                test.FixedCode = fixedSource;
+            }
+
             test.ExpectedDiagnostics.AddRange(expected);
+
             return test.RunAsync();
         }
+
+        // Retrieves the C# diagnostic for the specified rule, lines, columns, method and preferred method.
+        protected DiagnosticResult GetCSResultForRule(int startLine, int startColumn, int endLine, int endColumn, DiagnosticDescriptor rule, string methodName, string methodPreferredName)
+            => VerifyCS.Diagnostic(rule)
+                .WithSpan(startLine, startColumn, endLine, endColumn)
+                .WithArguments(methodName, methodPreferredName);
+
+        // Retrieves the VB diagnostic for the specified rule, lines, columns, method and preferred method.
+        protected DiagnosticResult GetVBResultForRule(int startLine, int startColumn, int endLine, int endColumn, DiagnosticDescriptor rule, string methodName, string methodPreferredName)
+            => VerifyVB.Diagnostic(rule)
+                .WithSpan(startLine, startColumn, endLine, endColumn)
+                .WithArguments(methodName, methodPreferredName);
+
+        protected string GetByteArrayWithDataCSharp(bool isEmpty) => isEmpty ? "" : @"byte[] buffer = { 0xBA, 0x5E, 0xBA, 0x11, 0xF0, 0x07, 0xBA, 0x11 };";
+
+        protected string GetByteArrayWithDataVisualBasic(bool isEmpty) => isEmpty ? "" : @"Dim buffer As Byte() = {&HBA, &H5E, &HBA, &H11, &HF0, &H07, &HBA, &H11}";
+
+        protected string GetByteArrayWithoutDataCSharp(bool isEmpty) => isEmpty ? "" : @"byte[] buffer = new byte[s.Length];";
+
+        protected string GetByteArrayWithoutDataVisualBasic(bool isEmpty) => isEmpty ? "" : @"Dim buffer As Byte() = New Byte(s.Length - 1) {}";
+
+        protected string GetConfigureAwaitCSharp(bool isEmpty) => isEmpty ? "" : @".ConfigureAwait(false)";
+
+        protected string GetConfigureAwaitVisualBasic(bool isEmpty) => isEmpty ? "" : @".ConfigureAwait(False)";
+
+        public static IEnumerable<object[]> CSharpUnnamedArgumentsTestData()
+        {
+            yield return new object[] { "buffer, 0, buffer.Length",
+                                        "buffer.AsMemory(0, buffer.Length)" };
+            yield return new object[] { "buffer, 0, buffer.Length, new CancellationToken()",
+                                        "buffer.AsMemory(0, buffer.Length), new CancellationToken()" };
+        }
+
+        public static IEnumerable<object[]> CSharpNamedArgumentsTestData()
+        {
+            // Normal argument order is: (byte[] buffer, int offset, int count)
```

Comment:
Nice extensive tests! Can we also add a test where some named arguments are followed by unnamed arguments? For example, a common case would be to only use named arguments for constant values.

```csharp
            yield return new object[] { "buffer, offset: 0, buffer.Length",
                                        "buffer.AsMemory(start: 0, buffer.Length)" };
```

Reply:
Done

Reply:
Added all the combinations of skipping names.

Reply:
I just realized the unit tests failed for one of these tests. Good catch, @mavasani. I'm working on getting it fixed.

Reply:
Nevermind, it was not a unit test failure caused by an unhandled case. It was a typo in one of my new testdata yielded objects. I fixed it, they are passing.